lib/hyperion/hyperion.rb

```diff
@@ -15,12 +15,13 @@ class Hyperion
   # @param body [String] the body to send with POST or PUT
   # @param additional_headers [Hash] headers to send in addition to the ones
   #   already determined by the route. Example: +{'User-Agent' => 'Mozilla/5.0'}+
+  # @param timeout [Integer] The limit of the entire request in seconds
   # @yield [result] yields the result if a block is provided
   # @yieldparam [HyperionResult]
   # @return [HyperionResult, Object] If a block is provided, returns the block's
   #   return value; otherwise, returns the result.
-  def self.request(route, body=nil, additional_headers={}, &block)
-    self.new(route).request(body, additional_headers, &block)
+  def self.request(route, body: nil, additional_headers: {}, timeout: 0, &block)
```

Comment:
There are no breaking call sites. Web, Quartermaster, Nexus, Contra, and Assaymatic do not use `Hyperion.request` directly. They go through `Hyperion::Requestor` instead.